vars/mtaBuild.groovy

```diff
@@ -61,10 +63,23 @@ void call(Map parameters = [:]) {
 
         dockerExecute(script: script, dockerImage: configuration.dockerImage, dockerOptions: configuration.dockerOptions) {
 
-            // Apply maven user-settings (for custom repositories, etc)
-            if (configuration.projectSettingsFile) {
+            def projectSettingsFile = configuration.projectSettingsFile
+            if (projectSettingsFile?.trim()) {
+                if(projectSettingsFile.trim().startsWith("http")){
+                    downloadSettingsFromUrl(projectSettingsFile)
+                    projectSettingsFile = "settings.xml"
+                }
                 sh 'mkdir -p $HOME/.m2'
-                sh "cp ${configuration.projectSettingsFile} \$HOME/.m2/settings.xml"
+                sh "cp ${projectSettingsFile} \$HOME/.m2/settings.xml"
+            }
+
+            def globalSettingsFile = configuration.globalSettingsFile
```

Comment:
I suggest trimming here so there is no need to trim twice below. Same applies for `def projectSettingsFile = ...` above.

Reply:
Agree, updated, thanks